### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by aloraps** [konsoletyper/teavm#957] [General]
Hello.

I am wondering if there is a roadmap for TeaVM available somewhere? I see a lot of progress on WASM which is exciting, it seems like some things that were previously unavailable like WASM-JS interoperability may soon be available.

Thanks!

**Comment by konsoletyper** [comment]
Here's mostly outdated list of features: https://github.com/konsoletyper/teavm/issues?q=is%3Aopen+is%3Aissue+label%3Aroadmap

In what aspect are you interested?

**Comment by aloraps** [comment]
Overall TeaVM has been wonderful for me when targeting JS, so I would say I just want to see how viable WASM target type may be in the future. I know that the GC is actively being worked on, but it would be nice to have a clearer understanding of what is left to do before WASM is a more stable target.

I also wanted to see if any JS performance optimizations may be on the horizon.

I also just increased my sponsorship, keep up the great work!

**Reply by konsoletyper** [reply]
> I would say I just want to see how viable WASM target type may be in the future

Here's a quick preview: https://teavm.org/gallery/jbox2d/index.html. I'm planning to post announcement soon. Please, wait a little bit.

> but it would be nice to have a clearer understanding of what is left to do before WASM is a more stable target

Frankly speaking, I don't know. To make Wasm GC BE stable, I need to find bugs there, and to find bugs I need volunteers who would use it for their projects and report some feedback.

> I also wanted to see if any JS performance optimizations may be on the horizon

Hm, I don't think there's much to do there. At least, nothing that's I'm aware of. If you know some certain performance issues, you can report them and I'll think what I can do there.

>  also just increased my sponsorship, keep up the great work!

Thanks!

**Reply by aloraps** [reply]
Great, looking forward to the announcement!

Btw, I tried the preview link on my iPhone, the WASM GC demo doesn't work but the other 3 work fine.

Also, I would be glad to test out the new backend as long as there is enough documentation for me to know how to target it, and as long as I have sufficient JS interop.

**Reply by konsoletyper** [reply]
> Btw, I tried the preview link on my iPhone, the WASM GC demo doesn't work but the other 3 work fine.

That's because Wasm GC does not work on Safari (Chrome under iOS is actually just wrapper around Safari, no other browser engines possible there). You can take someone's Android and test. See this chart: https://webassembly.org/features/

**Reply by aloraps** [reply]
I see, that makes sense. Looks like Safari is working on it and it's out in their latest Technology Preview.

Nonetheless, I would be glad to test it out once you make the announcement! Cheers.
